components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -3104,4 +3110,34 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
     }
     //
     //#endregion
+
+    /**
+     * This method is temporary until we moved image-builder into workspace clusters
+     * @param user
+     * @param workspace
+     * @param instance
+     * @returns
+     */
+    protected async getImageBuilderClient(user: User, workspace: Workspace, instance?: WorkspaceInstance) {
+        const teams = await this.teamDB.findTeamsByUser(user.id);
+        const isMovedImageBuilder = await getExperimentsClientForBackend().getValueAsync("movedImageBuilder", false, {
+            userId: user.id,
+            projectId: workspace.projectId,
+            teams,
+        });
+        log.info(
+            { userId: user.id, workspaceId: workspace.id, instanceId: instance?.id },
+            "image-builder in workspace cluster?",
+            {
+                userId: user.id,
+                projectId: workspace.projectId,
+                isMovedImageBuilder,
+            },
+        );
+        if (isMovedImageBuilder) {
+            return this.wsClusterImageBuilderClientProvider.getClient(user, workspace, instance);
```

Comment:
Mostly my lack of context on this area.

What is the lifecycle of this client? Is it possible that once the client is instantiated (given some feature flag config), it would leak to another request?

Reply:
Ah, good find :see_no_evil: Going to fix...